include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -361,10 +365,17 @@ struct __parallel_transform_reduce_impl
                         auto __local_idx = __item_id.get_local_id(0);
                         auto __group_idx = __item_id.get_group(0);
                         // 1. Initialization (transform part). Fill local memory
+                        _Size __n_items;
                         if (__is_first)
+                        {
                             __transform_pattern1(__item_id, __n, /*global_offset*/ (_Size)0, __temp_local, __rngs...);
+                            __n_items = __transform_pattern1.output_size(__n, __work_group_size);
```

Comment:
Should we do this calculation on the host instead?

Reply:
No, this must be in the device side as the `output_size` is calculated differently it the device code target is `SPIR`. This is only known during device compilation, and therefore must be in the kernel.

Reply:
That makes sense, thanks.